### PR TITLE
Establish IPFS connectivity for PubSub

### PR DESCRIPTION
**Pull request by ukstv**
Why manual connection? Apparently, it takes a lot of time for IPFS to connect to bootstrap nodes.

Related: #778

### CODE REVIEW

**Comment by oed** [comment]
> Why manual connection? Apparently, it takes a lot of time for IPFS to connect to bootstrap nodes.

Ok that's still the case -.- We saw the same thing happen with 3box way back.

File: packages/cli/src/build-ipfs-connection.util.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            config: {
+                Routing: {
+                    Type: IPFS_DHT_SERVER_MODE ? 'dhtserver' : 'dhtclient',
+                },
+                Bootstrap: BOOTSTRAP_LIST
+            }
+        })
+        const shouldConnect = network === "testnet-clay" || network === "mainnet"
+        if (shouldConnect) {
+            await Promise.all(BOOTSTRAP_LIST.map(async node => {
```
Comment (oed) [inline packages/cli/src/build-ipfs-connection.util.ts:54]: Do we really need to await this? I don't think this is blocking for the rest of the code?

File: packages/cli/src/build-ipfs-connection.util.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+const IPFS_DHT_SERVER_MODE = process.env.IPFS_DHT_SERVER_MODE === 'true'
+
+const BOOTSTRAP_LIST = [
+    '/dns4/ipfs-dev.3boxlabs.com/tcp/4012/wss/p2p/QmeRWpCaEpQLY1yJFFHf7pFbZZapXoj5E4pKvUtyExMUKk',
+    '/dns4/ipfs-dev.ceramic.network/tcp/4012/wss/p2p/QmadvKqYyLMHYjnwswd2bHM3UCFp4F4oSFdobYUKSPjBaZ',
+    '/dns4/ipfs-dev-cas.3boxlabs.com/tcp/4012/wss/p2p/QmPHLQoWhK4CMPPgxGQxjNYEp1fMB8NPpoLaaR2VDMNbcr',
+    '/dns4/ipfs-clay-cas.3boxlabs.com/tcp/4012/wss/p2p/QmQotCKxiMWt935TyCBFTN23jaivxwrZ3uD58wNxeg5npi',
+    '/dns4/ipfs-clay.ceramic.network/tcp/4012/wss/p2p/QmQqq1oVkwpJdKNdiWvrftgWE2nov4AFGPBauZ2AakxmoR',
+    '/dns4/ipfs-clay.3boxlabs.com/tcp/4012/wss/p2p/QmQotCKxiMWt935TyCBFTN23jaivxwrZ3uD58wNxeg5npi'
```
Comment (oed) [inline packages/cli/src/build-ipfs-connection.util.ts:22]: Can we make sure to connect to the clay nodes only when `network` is configured to `testnet-clay`?

File: packages/cli/src/build-ipfs-connection.util.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+multiformats.multicodec.add(dagJose)
+const dagJoseFormat = legacy(multiformats, dagJose.name)
+
+const IPFS_DHT_SERVER_MODE = process.env.IPFS_DHT_SERVER_MODE === 'true'
+
+const BOOTSTRAP_LIST = [
+    '/dns4/ipfs-dev.3boxlabs.com/tcp/4012/wss/p2p/QmeRWpCaEpQLY1yJFFHf7pFbZZapXoj5E4pKvUtyExMUKk',
+    '/dns4/ipfs-dev.ceramic.network/tcp/4012/wss/p2p/QmadvKqYyLMHYjnwswd2bHM3UCFp4F4oSFdobYUKSPjBaZ',
+    '/dns4/ipfs-dev-cas.3boxlabs.com/tcp/4012/wss/p2p/QmPHLQoWhK4CMPPgxGQxjNYEp1fMB8NPpoLaaR2VDMNbcr',
```
Comment (oed) [inline packages/cli/src/build-ipfs-connection.util.ts:19]: Not sure we want to connect to these. Are they also configured with Clay @valmack ?

File: packages/cli/src/build-ipfs-connection.util.ts (hunk 2 above)
Comment (oed) [reply packages/cli/src/build-ipfs-connection.util.ts:22]: Ah, see that you did something like this 👍 I imagine when we have the mainnet nodes up we will connect to a different set of nodes of that configuration :)

File: packages/cli/src/build-ipfs-connection.util.ts (hunk 1 above)
Comment (ukstv) [reply packages/cli/src/build-ipfs-connection.util.ts:54]: It's controversial. If you do not wait, the daemon _certainly_ proceeds with an empty set of peers. If you wait, it leaves you with the "best effort" connectivity, which most probably is all right for application to proceed.

File: packages/cli/src/build-ipfs-connection.util.ts (hunk 2 above)
Comment (ukstv) [reply packages/cli/src/build-ipfs-connection.util.ts:22]: So, as for now, all the nodes are still here @oed @valmack ?

Comment (ukstv) [reply packages/cli/src/build-ipfs-connection.util.ts:22]: Clarification regarding network names: if the "bootstrap" nodes do not follow some other pubsub topic like "/testing/bugging", the nodes connected to the bootstrap ones would not hear from the topic, unless finally connected to each other.

Comment (oed) [reply packages/cli/src/build-ipfs-connection.util.ts:22]: These three nodes we should connect to only if we are on `testnet-clay`.

File: packages/cli/src/build-ipfs-connection.util.ts (hunk 3 above)
Comment (oed) [reply packages/cli/src/build-ipfs-connection.util.ts:19]: These three nodes we should connect to only if we are on `dev-unstable`.

Comment (ukstv) [reply packages/cli/src/build-ipfs-connection.util.ts:19]: aha, clear

**Review by oed — APPROVED** [review]
lgtm

**Pull request merged**